spec/index.bs

```diff
@@ -873,6 +926,113 @@ interface GPUBufferUsage {
 
 ## Buffer Mapping ## {#buffer-mapping}
 
+An application can request to map a {{GPUBuffer}} to get its mapping which is
+an {{ArrayBuffer}} representing the {{GPUBuffer}}'s allocation. Mappings are
+requested asynchronously so that the user agent can ensure the GPU finished
+using the {{GPUBuffer}} before the application gets its mapping. Mappings can
+be requested for reading with {{GPUBuffer/mapReadAsync}} or writing with
+{{GPUBuffer/mapWriteAsync}}. A mapped {{GPUBuffer}} cannot be used by the GPU
+and must be unmapped using {{GPUBuffer/unmap}} before it can be used on the
+[=Queue timeline=].
+
+Issue: Add client-side validation that a mapped buffer can only be unmapped and destroyed on the worker on which it was mapped.
+
+### {{GPUBuffer/mapReadAsync|GPUDevice.mapReadAsync}} ### {#GPUBuffer-mapReadAsync}
+
+<dl dfn-type="method" dfn-for="GPUBuffer">
+    : <dfn>mapReadAsync()</dfn>
+    ::
+        <div algorithm="GPUBuffer.mapReadAsync()">
+
+            Issue: Handle error buffers once we have a description of the error monad.
+
+            1. If the {{[[usage]]}} slot of |this| doesn't contain the {{GPUBufferUsage/MAP_READ}} bit or
+                if {{[[state]]}} isn't [=buffer state/unmapped=]:
+
+                1. Record a validation error on the current scope.
+                1. Return a new promise that rejects.
```

Comment:
Can now use `[=a promise rejected with=]`

Reply:
Done, and chose `AbortError` for the rejection.